pkgs/tools/misc/toybox/default.nix

```diff
@@ -0,0 +1,67 @@
+{
+  stdenv, lib, fetchFromGitHub, buildPackages,
+  enableStatic ? false,
+  enableMininal ? false,
+  extraConfig ? ""
+}:
+stdenv.mkDerivation rec {
+  name = "toybox-${version}";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "landley";
+    repo = "toybox";
+    rev = "b11e1206b4193dd46b4c40c333175fceadd409df";
+    sha256 = "0pghxf2ihx79rmqqgg61850m6kcnck51nn1ng5x20vd0wfh6gdl6";
+  };
+
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+  buildInputs = lib.optionals enableStatic [ stdenv.cc.libc stdenv.cc.libc.static ];
+
+  postPatch = "patchShebangs scripts";
+
+  configurePhase = ''
+    make ${if enableMininal then "allnoconfig" else if stdenv.isFreeBSD then "freebsd_defconfig" else if stdenv.isDarwin then "macos_defconfig" else "defconfig"}
+
+    function parseconfig {
+      while read LINE; do
+        NAME="$(echo "$LINE" | sed -e 's/^\([^ =]*\).*$/\1/')"
+        OPTION="$(echo "$LINE" | sed -e 's/[^ =]*[ =]\([^ =]*\)$/\1/')"
+
+        if ! [[ "$NAME" =~ ^CONFIG_ ]]; then continue; fi
+
+        echo "parseconfig: removing $NAME"
+        sed -i /$NAME'\(=\| \)'/d .config
+
+        if [ "$OPTION" = 'n' ]; then
+          echo "parseconfig: unsetting $NAME"
+          echo "# $NAME is not set" >> .config
+        else
+          echo "parseconfig: setting $NAME=$OPTION"
+          echo "$NAME=$OPTION" >> .config
+        fi
+      done
+    }
+
+    cat << EOF | parseconfig
+    ${extraConfig}
+    EOF
+
+    make oldconfig
+  '';
+
+  makeFlags = [ "PREFIX=$(out)" ] ++ stdenv.lib.optional enableStatic "LDFLAGS=--static";
+
+  # tests currently (as of 0.8.0) get stuck in an infinite loop
```

Comment:
Can you open an upstream issue for this?

Reply:
Upstream fixed infinite loop [here](https://github.com/landley/toybox/commit/b928ec480cd73fd83511c0f5ca786d1b9f3167c3), but I have yet to try the tests with that update.

Reply:
That update indeed makes the tests work.